webpack.config.js

```diff
@@ -0,0 +1,281 @@
+    },
+    devtool: 'source-map',
+    externals: [
+        {
+            // Modules that cannot be webpack'ed, see https://webpack.js.org/configuration/externals/
+
+            // the vscode-module is created on-the-fly and must be excluded.
+            vscode: 'commonjs vscode',
+
+            // Out util/getCoreNodeModule.js file uses a dynamic require, so we'll just use it directly as a .js file.
+            // Note that the source is in .js and not .ts because we don't want webpack to depend on npm run build
+            // (except currently it's required for tests)
+            './getCoreNodeModule': 'commonjs getCoreNodeModule',
+
+            // Pull the rest automatically from externalModulesClosure
+            ...getExternalsEntries()
+        }
+    ],
+    plugins: [
+        // Clean the dist folder before webpacking
+        new CleanWebpackPlugin(
+            ['dist'],
+            {
+                root: __dirname,
+                verbose: true,
+            }),
+
+        // Copy files to dist folder where the runtime can find them
+        new CopyWebpackPlugin([
+            // Extension entrypoint
+            { from: './entrypoint.js' },
+
+            // getCoreNodeModule.js
+            { from: './utils/getCoreNodeModule.js', to: 'node_modules' },
+
+            // Images
+            { from: './images', to: 'images' },
+
+            // Test files
+            { from: './out/test', to: 'test' }
+        ]),
+
+        // External node modules
+        getExternalsCopyEntry(),
+
+        // vscode-languageserver/lib/files.js has one function which uses a dynamic require, but is not currently used by any dependencies
+        // Replace with a version that has only what is actually used.
+        new webpack.NormalModuleReplacementPlugin(
+            /[/\\]vscode-languageserver[/\\]lib[/\\]files\.js/,
+            require.resolve('./build/vscode-languageserver-files-stub.js')
+        ),
+
+        // Solve critical dependency issue in ./node_modules/ms-rest/lib/serviceClient.js (request of a dependency is an expression)
+        // for this line:
+        //
+        //   let data = require(packageJsonPath);
+        //
+        new webpack.ContextReplacementPlugin(
+            // Whenever there is a dynamic require that webpack can't analyze at all (i.e. resourceRegExp=/^\./), ...
+            /^\./,
+            (context) => {
+                // ... and the call was from within node_modules/ms-rest/lib...
+                if (/node_modules[/\\]ms-rest[/\\]lib/.test(context.context)) {
+                    // CONSIDER: Figure out how to make this work properly. The consequences of ignoring this error are that
+                    // the Azure SDKs (e.g. azure-arm-resource) don't get their info stamped into the user agent info for their calls.
+
+                    // // ... tell webpack that the call may be loading any of the package.json files from the 'node_modules/azure-arm*' folders
```

Comment:
Oh, it's quite confusing.  I forgot about this one, had trouble finding a proper fix, but the consequences aren't dire as noted. I'll enter as an engineering item.

Reply:
https://github.com/Microsoft/vscode-docker/issues/651

Reply:
Improved comments:

                // ... and the call was from within node_modules/ms-rest/lib...
                if (/node_modules[/\\]ms-rest[/\\]lib/.test(context.context)) {
                    /* CONSIDER: Figure out how to make this work properly.

                        // ... tell webpack that the call may be loading any of the package.json files from the 'node_modules/azure-arm*' folders
                        // so it will include those in the package to be available for lookup at runtime
                        context.request = path.resolve(__dirname, 'node_modules');
                        context.regExp = /azure-arm.*package\.json/;
                    */

                    // In the meantime, just ignore the error by telling webpack we've solved the critical dependency issue.
                    // The consequences of ignoring this error are that
                    //   the Azure SDKs (e.g. azure-arm-resource) don't get their info stamped into the user agent info for their calls.
                    for (const d of context.dependencies) {
                        if (d.critical) { d.critical = false; }
                    }
                }